### PR TITLE
Ask device permissions based on role

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

If role is Consumer, then only request for audio permissions. 

- Need to do access audio devices to show speaker options in dropdown. 
- Also we disable the 'Start Call' button in the configuration page if microphonePermissionGranted is false.

Request for audio and video permissions for other roles or no role at all.

Let me know if there are changes needed.

# Why

https://skype.visualstudio.com/SPOOL/_workitems/edit/2919171
https://skype.visualstudio.com/SPOOL/_workitems/edit/2919172

# How Tested

Tested on local rooms sample:
https://user-images.githubusercontent.com/79475487/179336360-790989e5-52b0-4dd6-8c63-68a690aa8088.mp4
Note: There is a ComplianceBanner showing camera device permissions are not granted for the Consumer role. This probably is confusing for the user. Let me know if I should remove this in this PR or another PR.

Also added unit tests

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by alkwa-msft — APPROVED** [review]

**Review by prabhjot-msft — APPROVED** [review]

File: packages/react-composites/src/composites/CallComposite/CallComposite.tsx [hunk 1]
```diff
@@ -194,12 +194,17 @@ export const CallComposite = (props: CallCompositeProps): JSX.Element => {
   } = props;
   useEffect(() => {
     (async () => {
-      await adapter.askDevicePermission({ video: true, audio: true });
-      adapter.queryCameras();
-      adapter.queryMicrophones();
+      if (role === 'Consumer') {
+        // Need to ask for audio devices to get access to speakers
```
Comment (PorterNan) [inline packages/react-composites/src/composites/CallComposite/CallComposite.tsx:198]: Do we need this? I don't think access to speaker need any permissions? I don't remember youtube is asking me any permissions to play a video

File: samples/Calling/src/app/utils/AppUtils.ts [hunk 2]
```diff
@@ -59,7 +60,7 @@ export const createRoomId = async (): Promise<string> => {
  * Joins an ACS room with a given roomId and role
  */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export const joinRoom = async (userId: string, roomId: string, role): Promise<void> => {
+export const joinRoom = async (userId: string, roomId: string, role: Role): Promise<void> => {
```
Comment (alkwa-msft) [inline samples/Calling/src/app/utils/AppUtils.ts:63]: do we need to conditional hide this as well?

File: packages/react-composites/src/composites/CallComposite/CallComposite.tsx (hunk 1 above)
Comment (mgamis-msft) [reply packages/react-composites/src/composites/CallComposite/CallComposite.tsx:198]: I will look into how to get speakers access without permissions from the Calling SDK.

File: samples/Calling/src/app/utils/AppUtils.ts (hunk 2 above)
Comment (mgamis-msft) [reply samples/Calling/src/app/utils/AppUtils.ts:63]: Yes when we merge it into main.

**Review by PorterNan — APPROVED** [review]

**Pull request merged**